### PR TITLE
develop: improve python mysqlclient sample code doc

### PR DESCRIPTION
**Pull request by Mini256**
### First-time contributors' checklist 

- [x] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs-cn) that's required for repo owners to accept my contribution.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions (in Chinese)](https://github.com/pingcap/docs-cn/blob/master/CONTRIBUTING.md#版本选择指南).

- [x] master (the latest development version)
- [ ] v7.4 (TiDB 7.4 versions)
- [ ] v7.3 (TiDB 7.3 versions)
- [ ] v7.2 (TiDB 7.2 versions)
- [x] v7.1 (TiDB 7.1 versions)
- [ ] v7.0 (TiDB 7.0 versions)
- [x] v6.5 (TiDB 6.5 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs/pull/14544
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

**Comment by Oreoxmt** [comment]
@Mini256 https://github.com/Mini256/docs-cn/pull/1 PTAL

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 1]
```diff
@@ -27,14 +24,18 @@ TiDB 是一个兼容 MySQL 的数据库。[mysqlclient](https://pypi.org/project
...
 ```
 
-### 第 2 步：安装依赖 (包括 mysqlclient)
+### 第 2 步：安装依赖
+
+运行以下命令，安装示例代码所需要的依赖 (包括 mysqlclient)：
 
 ```bash
-cd tidb-example-python;
 pip install -r requirement.txt
```
Comment (wd0517) [inline develop/dev-guide-sample-application-python-mysqlclient.md:39]:
```suggestion
pip install -r requirements.txt
```

**Review by wd0517 — APPROVED** [review]

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 2]
```diff
@@ -27,17 +31,23 @@ TiDB 是一个兼容 MySQL 的数据库。[mysqlclient](https://pypi.org/project
...
 ```bash
-git clone https://github.com/pingcap-inc/tidb-example-python.git
+git clone https://github.com/tidb-samples/tidb-python-mysqlclient-quickstart.git
+cd tidb-python-mysqlclient-quickstart
 ```
 
-### 第 2 步：安装依赖 (包括 mysqlclient)
+### 第 2 步：安装依赖
+
+运行以下命令，安装示例代码所需要的依赖 (包括 mysqlclient)：
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:43]:
```suggestion
运行以下命令，安装示例代码所需要的依赖（包括 mysqlclient）：
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 3]
```diff
@@ -148,16 +164,18 @@ pip install -r requirement.txt
...
 1. 运行下述命令，执行示例代码：
 
-    ```python
-    python3 mysqlclient_example.py
+    ```bash
+    python mysqlclient_example.py
     ```
 
-2. 查看[示例输出](https://github.com/pingcap-inc/tidb-example-python/blob/main/Expected-Output.md#mysqlclient)，并与你的程序输出进行比较。结果近似即为连接成功。
+2. 查看[`Expected-Output.txt`](https://github.com/tidb-samples/tidb-python-mysqlclient-quickstart/blob/main/Expected-Output.txt)，并与你的程序输出进行比较。结果近似即为连接成功。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:171]:
```suggestion
2. 查看 [`Expected-Output.txt`](https://github.com/tidb-samples/tidb-python-mysqlclient-quickstart/blob/main/Expected-Output.txt)，并与你的程序输出进行比较。结果近似即为连接成功。
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 4]
```diff
@@ -3,19 +3,23 @@ title: 使用 mysqlclient 连接到 TiDB
 summary: 本文描述了 TiDB 和 mysqlclient 的连接步骤，并给出了简单示例代码片段。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:3]:
```suggestion
summary: 了解如何使用 mysqlclient 连接到 TiDB。本文提供了使用 mysqlclient 与 TiDB 交互的 Python 示例代码片段。
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 5]
```diff
@@ -46,29 +56,35 @@ pip install -r requirement.txt
...
-1. 在 TiDB Cloud Web Console 中，选择你的 TiDB Serverless 集群，进入 **Overview** 页面，点击右上角的 **Connect** 按钮。
+1. 在 TiDB Cloud 的 [**Clusters**](https://tidbcloud.com/console/clusters) 页面中，选择你的 TiDB Serverless 集群，进入集群的 **Overview** 页面。
 
-2. 确认窗口中的配置和你的运行环境一致。
+2. 点击右上角的 **Connect** 按钮，将会弹出连接对话框。
 
-    - Endpoint 为 **Public**
-    - Connect With 选择 **General**
-    - Operating System 为你的运行环境。
+3. 确认窗口中的配置和你的运行环境一致。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:63]:
```suggestion
3. 确认对话框中的配置和你的运行环境一致。
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 6]
```diff
@@ -46,29 +56,35 @@ pip install -r requirement.txt
...
+    > 如果你之前已经生成过密码，可以直接使用原密码，或点击 **Reset password** 重新生成密码。
+
+5. 运行以下命令，将 `.env.example` 复制并重命名为 `.env`：
 
     ```bash
     cp .env.example .env
     ```
 
-5. 复制并粘贴对应连接字符串至 `.env` 中。示例结果如下。
+6. 复制并粘贴对应连接字符串至 `.env` 中。示例结果如下。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:85]:
```suggestion
6. 复制并粘贴对应连接字符串至 `.env` 中。示例结果如下：
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 7]
```diff
@@ -128,7 +144,7 @@ pip install -r requirement.txt
 
 2. 复制并粘贴对应 TiDB 的连接字符串至 `.env` 中。示例结果如下。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:145]:
```suggestion
2. 复制并粘贴对应 TiDB 的连接字符串至 `.env` 中。示例结果如下：
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 8]
```diff
@@ -219,18 +240,17 @@ with get_mysqlclient_connection(autocommit=True) as conn:
...
-        cursor.execute("DELETE FROM player WHERE id = %s", (player_id,))
+        cursor.execute("DELETE FROM players WHERE id = %s", (player_id,))
 ```
 
 更多信息参考[删除数据](/develop/dev-guide-delete-data.md)。
 
 ## 注意事项
 
-- 完整代码及其运行方式，见 [tidb-example-python](https://github.com/pingcap-inc/tidb-example-python/blob/main/README-zh.md) GitHub 仓库。
 - Driver 封装程度较低，因此在程序内会见到大量的 SQL 语句。与 ORM 不同，因为没有数据对象的存在，`mysqlclient` 的查询对象以元组 (tuple) 进行表示。虽然 Python 的 Driver 相较其他语言的使用更方便，但因其不可屏蔽底层实现，需手动管控事务的特性，如果没有大量必须使用 SQL 的场景，仍然推荐使用 ORM 进行程序编写。这可以降低程序的耦合性。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:250]:
```suggestion
### 使用驱动程序还是 ORM 框架？

Python 驱动程序提供对数据库的底层访问，但要求开发者：

- 手动建立和释放数据库连接
- 手动管理数据库事务
- 手动将数据行（在 mysqlclient 中表示为元组 (tuple) ）映射为数据对象

除非需要编写复杂的 SQL 语句，否则建议使用 [ORM](https://zh.wikipedia.org/wiki/对象关系映射) 框架进行开发，例如 [SQLAlchemy](/develop/dev-guide-sample-application-python-sqlalchemy.md)、[Peewee](/develop/dev-guide-sample-application-python-peewee.md) 和 Django。ORM 可以帮助你：

- 减少管理连接和事务的[模板代码](https://en.wikipedia.org/wiki/Boilerplate_code)
- 使用数据对象代替大量 SQL 语句来操作数据
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 9]
```diff
@@ -219,18 +240,17 @@ with get_mysqlclient_connection(autocommit=True) as conn:
...
-- 关于 mysqlclient 的更多使用方法，可以参考 [mysqlclient 官方文档](https://mysqlclient.readthedocs.io/)。
 
 ## 下一步
 
-- 你可以继续阅读开发者文档，以获取更多关于 TiDB 的开发者知识。例如：[插入数据](/develop/dev-guide-insert-data.md)，[更新数据](/develop/dev-guide-update-data.md)，[删除数据](/develop/dev-guide-delete-data.md)，[单表读取](/develop/dev-guide-get-data-from-single-table.md)，[事务](/develop/dev-guide-transaction-overview.md)，[SQL 性能优化](/develop/dev-guide-optimize-sql-overview.md)等。
+- 关于 mysqlclient 的更多使用方法，可以参考 [mysqlclient 官方文档](https://mysqlclient.readthedocs.io/)。
+- 你可以继续阅读开发者文档，以获取更多关于 TiDB 应用开发的最佳实践。例如：[插入数据](/develop/dev-guide-insert-data.md)，[更新数据](/develop/dev-guide-update-data.md)，[删除数据](/develop/dev-guide-delete-data.md)，[单表读取](/develop/dev-guide-get-data-from-single-table.md)，[事务](/develop/dev-guide-transaction-overview.md)，[SQL 性能优化](/develop/dev-guide-optimize-sql-overview.md)等。
 - 如果你更倾向于参与课程进行学习，我们也提供专业的 [TiDB 开发者课程](https://cn.pingcap.com/courses-catalog/back-end-developer/?utm_source=docs-cn-dev-guide)支持，并在考试后提供相应的[资格认证](https://learn.pingcap.com/learner/certification-center)。
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:256]: @Mini256 Please confirm whether the **Need help** section needs to be added to the Chinese document
```
## Need help?

Ask questions on the [Discord](https://discord.gg/vYU9h56kAX), or [create a support ticket](https://support.pingcap.com/).
```

**Review by Oreoxmt — COMMENTED** [review]
Rest LGTM

File: develop/dev-guide-sample-application-python-mysqlclient.md (hunk 9 above)
Comment (sykp241095) [reply develop/dev-guide-sample-application-python-mysqlclient.md:267]: Use asktug.com if it's Chinese version.

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 10]
```diff
@@ -46,29 +56,35 @@ pip install -r requirement.txt
...
 
-3. 点击 **Generate Password** 生成密码。
-   
-   <Tip>如果你之前已经生成过密码，可以直接使用原密码，或点击 **Reset Password** 重新生成密码。</Tip>
+    > **Tip:**
+    >
+    > 如果你在 Windows Subsystem for Linux (WSL) 中运行，请切换为对应的 Linux 发行版。
 
-4. 运行以下命令，将 `.env.example` 复制并重命名为 `.env`：
+4. 如果你还设置密码，点击 **Create password** 生成一个随机密码。
```
Comment (ran-huang) [inline develop/dev-guide-sample-application-python-mysqlclient.md:73]:
```suggestion
4. 如果你还没有设置密码，点击 **Create password** 生成一个随机密码。
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 11]
```diff
@@ -219,18 +240,28 @@ with get_mysqlclient_connection(autocommit=True) as conn:
...
-- 关于 mysqlclient 的更多使用方法，可以参考 [mysqlclient 官方文档](https://mysqlclient.readthedocs.io/)。
+### 使用驱动程序还是 ORM 框架？
+
+Python 驱动程序提供对数据库的底层访问，但要求开发者：
+
+- 手动建立和释放数据库连接
+- 手动管理数据库事务
+- 手动将数据行（在 mysqlclient 中表示为元组 (tuple) ）映射为数据对象
+
+除非需要编写复杂的 SQL 语句，否则建议使用 [ORM](https://zh.wikipedia.org/wiki/对象关系映射) 框架进行开发，例如 [SQLAlchemy](/develop/dev-guide-sample-application-python-sqlalchemy.md)、[Peewee](/develop/dev-guide-sample-application-python-peewee.md) 和 Django。ORM 可以帮助你：
```
Comment (ran-huang) [inline develop/dev-guide-sample-application-python-mysqlclient.md:258]: “除非...否则”感觉有点绕
```suggestion
建议仅在需要编写复杂的 SQL 语句时使用驱动程序。其他情况下，建议使用 [ORM](https://zh.wikipedia.org/wiki/对象关系映射) 框架进行开发，例如 [SQLAlchemy](/develop/dev-guide-sample-application-python-sqlalchemy.md)、[Peewee](/develop/dev-guide-sample-application-python-peewee.md) 和 Django。ORM 可以帮助你：
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 12]
```diff
@@ -219,18 +240,28 @@ with get_mysqlclient_connection(autocommit=True) as conn:
...
+
+- 减少管理连接和事务的[模板代码](https://en.wikipedia.org/wiki/Boilerplate_code)
+- 使用数据对象代替大量 SQL 语句来操作数据
 
 ## 下一步
 
-- 你可以继续阅读开发者文档，以获取更多关于 TiDB 的开发者知识。例如：[插入数据](/develop/dev-guide-insert-data.md)，[更新数据](/develop/dev-guide-update-data.md)，[删除数据](/develop/dev-guide-delete-data.md)，[单表读取](/develop/dev-guide-get-data-from-single-table.md)，[事务](/develop/dev-guide-transaction-overview.md)，[SQL 性能优化](/develop/dev-guide-optimize-sql-overview.md)等。
+- 关于 mysqlclient 的更多使用方法，可以参考 [mysqlclient 官方文档](https://mysqlclient.readthedocs.io/)。
+- 你可以继续阅读开发者文档，以获取更多关于 TiDB 应用开发的最佳实践。例如：[插入数据](/develop/dev-guide-insert-data.md)，[更新数据](/develop/dev-guide-update-data.md)，[删除数据](/develop/dev-guide-delete-data.md)，[单表读取](/develop/dev-guide-get-data-from-single-table.md)，[事务](/develop/dev-guide-transaction-overview.md)，[SQL 性能优化](/develop/dev-guide-optimize-sql-overview.md)等。
```
Comment (ran-huang) [inline develop/dev-guide-sample-application-python-mysqlclient.md:266]:
```suggestion
- 你可以继续阅读开发者文档，以获取更多关于 TiDB 应用开发的最佳实践。例如：[插入数据](/develop/dev-guide-insert-data.md)、[更新数据](/develop/dev-guide-update-data.md)、[删除数据](/develop/dev-guide-delete-data.md)、[单表读取](/develop/dev-guide-get-data-from-single-table.md)、[事务](/develop/dev-guide-transaction-overview.md)、[SQL 性能优化](/develop/dev-guide-optimize-sql-overview.md)等。
```

File: develop/dev-guide-sample-application-python-mysqlclient.md (hunk 11 above)
Comment (Mini256) [reply develop/dev-guide-sample-application-python-mysqlclient.md:258]: @Oreoxmt 看看呢

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 13]
```diff
@@ -219,18 +240,28 @@ with get_mysqlclient_connection(autocommit=True) as conn:
...
-- 完整代码及其运行方式，见 [tidb-example-python](https://github.com/pingcap-inc/tidb-example-python/blob/main/README-zh.md) GitHub 仓库。
-- Driver 封装程度较低，因此在程序内会见到大量的 SQL 语句。与 ORM 不同，因为没有数据对象的存在，`mysqlclient` 的查询对象以元组 (tuple) 进行表示。虽然 Python 的 Driver 相较其他语言的使用更方便，但因其不可屏蔽底层实现，需手动管控事务的特性，如果没有大量必须使用 SQL 的场景，仍然推荐使用 ORM 进行程序编写。这可以降低程序的耦合性。
-- 关于 mysqlclient 的更多使用方法，可以参考 [mysqlclient 官方文档](https://mysqlclient.readthedocs.io/)。
+### 使用驱动程序还是 ORM 框架？
+
+Python 驱动程序提供对数据库的底层访问，但要求开发者：
+
+- 手动建立和释放数据库连接
+- 手动管理数据库事务
+- 手动将数据行（在 mysqlclient 中表示为元组 (tuple) ）映射为数据对象
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:256]:
```suggestion
- 手动将数据行（在 mysqlclient 中表示为元组 (tuple)）映射为数据对象
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 14]
```diff
@@ -77,33 +93,33 @@ pip install -r requirement.txt
...
     ```bash
     cp .env.example .env
     ```
 
-3. 复制并粘贴对应的连接字符串至 `.env` 中。示例结果如下：
+5. 复制并粘贴对应的连接字符串至 `.env` 中。示例结果如下：
 
-    ```python
+    ```dotenv
     TIDB_HOST='{host}.clusters.tidb-cloud.com'
```
Comment (Mini256) [inline develop/dev-guide-sample-application-python-mysqlclient.md:123]:
```suggestion
    TIDB_HOST='{host}'
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 15]
```diff
@@ -77,58 +93,58 @@ pip install -r requirement.txt
...
     ```bash
     cp .env.example .env
     ```
 
-2. 复制并粘贴对应 TiDB 的连接字符串至 `.env` 中。示例结果如下。
+2. 复制并粘贴对应 TiDB 的连接字符串至 `.env` 中。示例结果如下：
 
-    ```python
+    ```dotenv
     TIDB_HOST='{tidb_server_host}'
```
Comment (Mini256) [inline develop/dev-guide-sample-application-python-mysqlclient.md:148]:
```suggestion
    TIDB_HOST='{host}'
```

**Comment by Oreoxmt** [comment]
@ran-huang PTAL again

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 16]
```diff
@@ -46,29 +56,35 @@ pip install -r requirement.txt
...
     ```
 
-5. 复制并粘贴对应连接字符串至 `.env` 中。示例结果如下。
+6. 复制并粘贴对应连接字符串至 `.env` 中。示例结果如下：
 
-    ```python
+    ```dotenv
     TIDB_HOST='{gateway-region}.aws.tidbcloud.com'
     TIDB_PORT='4000'
     TIDB_USER='{prefix}.root'
```
Comment (Oreoxmt) [inline develop/dev-guide-sample-application-python-mysqlclient.md:90]: @Mini256 https://github.com/pingcap/docs-cn/pull/14843#discussion_r1309899060
```
    TIDB_HOST='{host}'
    TIDB_PORT='4000'
    TIDB_USER='{user}'
    
```

Comment (Mini256) [inline develop/dev-guide-sample-application-python-mysqlclient.md:90]:
```suggestion
    TIDB_HOST='{host}'  # e.g. gateway01.ap-northeast-1.prod.aws.tidbcloud.com
    TIDB_PORT='4000'
    TIDB_USER='{user}'  # e.g. xxxxxx.root
```

File: develop/dev-guide-sample-application-python-mysqlclient.md [hunk 17]
```diff
@@ -77,59 +93,59 @@ pip install -r requirement.txt
...
 
-3. 复制并粘贴对应的连接字符串至 `.env` 中。示例结果如下：
+5. 复制并粘贴对应的连接字符串至 `.env` 中。示例结果如下：
 
-    ```python
-    TIDB_HOST='{host}.clusters.tidb-cloud.com'
+    ```dotenv
+    TIDB_HOST='{host}'
     TIDB_PORT='4000'
     TIDB_USER='{username}'
```
Comment (Mini256) [inline develop/dev-guide-sample-application-python-mysqlclient.md:125]:
```suggestion
    TIDB_HOST='{host}'  # e.g. tidb.xxxx.clusters.tidb-cloud.com
    TIDB_PORT='4000'
    TIDB_USER='{user}'  # e.g. root
```

**Review by Oreoxmt — APPROVED** [review]

**Review by Oreoxmt — APPROVED** [review]

**Review by ran-huang — APPROVED** [review]

**Pull request merged**